bears/general/IndentationBear.py

```diff
@@ -0,0 +1,206 @@
+
+            if(something)
+            does not get indented
+
+        undergoes no change.
+
+        :param language_family:
+            Language family to be used for indentation.
+        :param language:
+            Language to be used for indentation.
+        :param use_spaces:
+            Insert spaces instead of tabs for indentation.
+        :param tab_width:
+            No. of spaces to insert for indentation.
+            Only Applicable if use_spaces is False
+        """
+        lang_settings_dict = LanguageDefinition(language, language_family)
+        annotation_dict = dependency_results[AnnotationBear.name][0].contents
+        indent_types = dict(lang_settings_dict["indent_types"])
+
+        try:
+            indent_levels = self.get_indent_levels(
+                                file, filename, indent_types, annotation_dict)
+
+        # This happens only in case of unmatched indents.
+        except UnmatchedIndentError as e:
+            yield Result(self, str(e), severity=RESULT_SEVERITY.MAJOR)
+            return
+
+        insert = ' '*tab_width if use_spaces else '\t'
+        new_file = []
+        for line_nr, line in enumerate(file):
+            # Leave out empty lines
+            if line.lstrip() != '':
+                new_file.append(insert*indent_levels[line_nr] +
+                                line.lstrip())
+            else:
+                new_file.append(line)
+
+        if new_file != list(file):
+            wholediff = Diff.from_string_arrays(file, new_file)
+            for diff in wholediff.split_diff():
+                yield Result(
+                    self,
+                    'The indentation could be changed to improve readability.',
+                    severity=RESULT_SEVERITY.INFO,
+                    affected_code=(diff.range(filename),),
+                    diffs={filename: diff})
+
+    def get_indent_levels(self, file, filename, indent_types, annotation_dict):
+        """
+        Gets the level of indentation of each line.
+
+        :param indent_types:    A dictionary with keys as start of indent and
+                                values as their corresponding closing indents.
+        :param annotation_dict: A dictionary containing sourceranges of all the
+                                strings and comments within a file.
+        :return:                A tuple containing the levels of indentation of
+                                each line.
+        """
+        indent_levels = []
+        ranges = self.get_indentation_sourcerange(
+                         file, filename, indent_types, annotation_dict)
+        indent, next_indent = 0, 0
+        for line in range(0, len(file)):
+            indent = next_indent
+            for _range in ranges:
+                if _range.start.line == line + 1:
+                    next_indent += 1
+                if(_range.end.line == line + 1 and
+                   (file[line].lstrip()[0] in indent_types.values())):
```

Comment:
its actually like 
```
if(something){
    code
}
```
and 
```
if(something){
    code}
```

are fine but 

```
if(something){
    code
    }
``` 
isn't it's actually loosely based on the [K&R](https://en.wikipedia.org/wiki/Indent_style#Tabs.2C_spaces.2C_and_size_of_indent) style indent 
though i plan to have it configurable later on